### PR TITLE
Improve Head component

### PR DESCRIPTION
**Pull request by JoaoVictor6**
Hello! Im João([@PrazerJo](https://twitter.com/PrazerJo)). I saw this project on my feed and I was interested. Here are my 5 cents on this project. I hope I can contribute more :)
## What problem the PR solves?
When reading this component, I see some problems with performance and readability. When a state is updated, the Component where this state is initialized has to render, this makes the component render two times always. How did I make resolve this? simple: [You do Not Need an Effect](https://react.dev/learn/you-might-not-need-an-effect).

## What solution the PR gives?
I removed the useEffect and unnecessary state, Keep It Simple

### CODE REVIEW

**Review by nidib — CHANGES_REQUESTED** [review]
Really nice PR. The useEffect is really not necessary.

Just these few considerations for it to be a go on my end

File: src/components/Head/Head.tsx [hunk 1]
```diff
@@ -1,25 +1,20 @@
...
+  const enableTasks = tasks.filter((task) => Boolean(task))
+  const titlePrefix= enableTasks.length ? `[${enableTasks.length}]` : undefined
+  const defaultTitle = "phived - the anti-procrastination to-do list"
 
   return (
     <Helmet>
-      <title>{title}</title>
-      <link rel="icon" type="image/x-icon" href={icon} />
+      <title>{
+        [titlePrefix, defaultTitle].join(' ')
```
[the remark this answers is not in the document]
Comment (JoaoVictor6) [reply src/components/Head/Head.tsx:17]: Make more sense. I was very sleepy when made 😆

File: src/components/Head/Head.tsx [hunk 2]
```diff
@@ -1,28 +1,25 @@
...
-  }, [tasks]);
+  const enableTasks = tasks.filter(Boolean)
+  const titlePrefix= enableTasks.length ? `[${enableTasks.length}]` : ''
 
   return (
     <Helmet>
-      <title>{title}</title>
-      <link rel="icon" type="image/x-icon" href={icon} />
+      <title>{
+        `${titlePrefix} ${BASE_PAGE_TITLE}`
```
Comment (nidib) [inline src/components/Head/Head.tsx:16]: #review
- if there is no title prefix, it'll have an empty character before the title. We need to trim this string.

#suggestion
- My suggestion to keep it bit cleaner:
```js
const title = `${titlePrefix} ${BASE_PAGE_TITLE}`.trim();
const iconPath = `/${enableTasks.length ? ALERT_ICON : DEFAULT_ICON}`;
```

```jsx
<title>{title}</title>
<link 
        rel="icon" 
        type="image/x-icon" 
        href={iconPath} 
/>
```

**Review by nidib — CHANGES_REQUESTED** [review]

**Comment by Nick-Gabe** [comment]
@JoaoVictor6 Please click on "resolve" when you resolve any comment, just for us to know what you already fixed or not, thanks :)

File: src/components/Head/Head.tsx [hunk 3]
```diff
@@ -1,28 +1,25 @@
 import { Helmet } from "react-helmet-async";
-import { useEffect, useState } from "react";
 import { useTasksContext } from "src/contexts";
 
+const ALERT_ICON = "favicon-alert.ico"
+const DEFAULT_ICON = "favicon-default.ico"
+const BASE_PAGE_TITLE = "phived - the anti-procrastination to-do list"
```
Comment (Nick-Gabe) [inline src/components/Head/Head.tsx:6]: I suggest putting those in a separate file, for example
```typescript
// headConstants.ts
export const headConstants = {
  icons: {
    default: "favicon-default.ico",
    alert: "favicon-alert.ico"
  },
  baseTitle: "phived - the anti-procrastination to-do list"
}
```

File: src/components/Head/Head.tsx [hunk 4]
```diff
@@ -1,28 +1,25 @@
...
-      setIcon("/favicon-alert.ico");
-      setTitle(
-        `[${tasks.filter((task) => !!task).length}] phived · the anti-procrastination to-do list`
-      );
-    } else {
-      setIcon("/favicon-default.ico");
-      setTitle("phived · the anti-procrastination to-do list");
-    }
-  }, [tasks]);
+  const enableTasks = tasks.filter(Boolean)
```
Comment (Nick-Gabe) [inline src/components/Head/Head.tsx:10]: Maybe renaming this variable to "pendingTasks"

File: src/components/Head/Head.tsx (hunk 2 above)
Comment (Nick-Gabe) [reply src/components/Head/Head.tsx:16]: There's no need to use Javascript template string here inside the title, one solution is to do as @nidib said

**Review by Nick-Gabe — CHANGES_REQUESTED** [review]

File: src/components/Head/Head.tsx (hunk 3 above)
Comment (JoaoVictor6) [reply src/components/Head/Head.tsx]: made :) I like to use constants' names in uppercase by convention, but I don't know if it is valid for this project

File: src/components/Head/Head.tsx (hunk 4 above)
Comment (JoaoVictor6) [reply src/components/Head/Head.tsx]: done :)

File: src/components/Head/Head.tsx (hunk 2 above)
Comment (JoaoVictor6) [reply src/components/Head/Head.tsx]: done :)

**Comment by JoaoVictor6** [comment]
> Thank you for addressing the changes, everything good! :D

It was really cool, I hope I can contribute more :)

File: src/components/Head/Head.tsx [hunk 5]
```diff
@@ -1,28 +1,23 @@
 import { Helmet } from "react-helmet-async";
-import { useEffect, useState } from "react";
 import { useTasksContext } from "src/contexts";
+import { headConstants } from "./headConstants";
```
Comment (nidib) [inline src/components/Head/Head.tsx:3]: we still don't have a github workflow to run linting and soon to have tests. But if you run `npm run lint`it should throw an error due to this relative import.

I suggest to run the lint and fix possible errors

**Review by nidib — CHANGES_REQUESTED** [review]

File: src/components/Head/Head.tsx (hunk 5 above)
Comment (JoaoVictor6) [reply src/components/Head/Head.tsx:3]: image
I fixed that, but the lint script is broken :/

Comment (nidib) [reply src/components/Head/Head.tsx:3]: Just checking, but did you npm install everything? There were lib updates in between PRs

Comment (JoaoVictor6) [reply src/components/Head/Head.tsx]: with `npm ci` the script worked!

**Review by nidib — APPROVED** [review]
lgtm

**Review by LukeberryPi — APPROVED** [review]
lgtm

**Pull request merged**